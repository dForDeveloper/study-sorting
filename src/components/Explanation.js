import React from 'react';
import '../styles/main.scss';

function Explanation(props) {
  // const { boxIds, action, i , algorithmName } = props;
  let explanation = '';
  return (
    <div className="explanation">
      <p>
        <span>{explanation}</span>
      </p>
    </div>
  )
  // if (action === 'swap' && algorithmName === 'Bubble Sort') {
  //   return (
  //     <div className="explanation">
  //       <p>
  //         <span>{boxIds[i]}</span> swaps with <span>{boxIds[i + 1]}</span>
  //       </p>
  //     </div>
  //   );
  // } else if (action === 'compare' && boxIds[i] < boxIds[i + 1] &&
  //     algorithmName === 'Bubble Sort') {
  //   return (
  //     <div className="explanation">
  //       <p>
  //         compare <span>{boxIds[i]}</span> and <span>{boxIds[i + 1]}</span>
  //       </p>
  //       <p>they are in order</p>
  //     </div>
  //   );
  // } else if (action === 'compare' && algorithmName === 'Bubble Sort') {
  //   return (
  //     <div className="explanation">
  //       <p>
  //         compare <span>{boxIds[i]}</span> and <span>{boxIds[i + 1]}</span>
  //       </p>
  //       <p>they are out of order</p>
  //     </div>
  //   );
  // } else if (algorithmName === 'Insertion Sort') {
  //   return (
  //     <div className="explanation">
  //       <p>insertion sort explanation</p>
  //     </div>
  //   )
  // }
}

export default Explanation;