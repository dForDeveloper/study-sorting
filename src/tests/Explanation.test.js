import React from 'react';
import Explanation from '../components/Explanation';
import { shallow } from 'enzyme';

describe('Explanation', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(
      <Explanation
        boxIds={[1, 2, 3 , 4, 5, 6, 7, 8]}
        action="compare"
        i={0}
      />
    );
  });
});