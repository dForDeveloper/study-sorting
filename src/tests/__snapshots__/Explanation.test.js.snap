// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explanation should match snapshot for "compare" case 1`] = `
<div
  className="explanation"
>
  <p>
    compare 
    <span>
      8
    </span>
     and 
    <span>
      7
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "compare-adjacent" case 1`] = `
<div
  className="explanation"
>
  <p>
    compare 
    <span>
      7
    </span>
     and 
    <span>
      8
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "compare-again" case 1`] = `
<div
  className="explanation"
>
  <p>
    compare 
    <span>
      7
    </span>
     and 
    <span>
      5
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "end" case 1`] = `
<div
  className="explanation"
>
  <p>
    the array is sorted
  </p>
</div>
`;

exports[`Explanation should match snapshot for "examine" case 1`] = `
<div
  className="explanation"
>
  <p>
    compare 
    <span>
      8
    </span>
     with the elements to its left
  </p>
</div>
`;

exports[`Explanation should match snapshot for "greater-first-comparison" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      7
    </span>
     is greater than 
    <span>
      5
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "greater-multiple-comparisons" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      7
    </span>
     is greater than 
    <span>
      5
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "insert" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      5
    </span>
     gets inserted
  </p>
</div>
`;

exports[`Explanation should match snapshot for "less-than-all" case 1`] = `
<div
  className="explanation"
>
  <p>
    there are no numbers to compare with 
    <span>
      5
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "nothing-on-left" case 1`] = `
<div
  className="explanation"
>
  <p>
    there are no numbers to compare with 
    <span>
      5
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "shift" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      7
    </span>
     shifts to the right
  </p>
</div>
`;

exports[`Explanation should match snapshot for "sorted" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      8
    </span>
     and 
    <span>
      7
    </span>
     are in the correct order
  </p>
</div>
`;

exports[`Explanation should match snapshot for "stop-first-comparison" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      7
    </span>
     and 
    <span>
      5
    </span>
     are in the correct order
  </p>
</div>
`;

exports[`Explanation should match snapshot for "stop-multiple-comparisons" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      7
    </span>
     and 
    <span>
      5
    </span>
     are in the correct order
  </p>
</div>
`;

exports[`Explanation should match snapshot for "swap" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      8
    </span>
     swaps with 
    <span>
      7
    </span>
  </p>
</div>
`;

exports[`Explanation should match snapshot for "unsorted" case 1`] = `
<div
  className="explanation"
>
  <p>
    <span>
      8
    </span>
     and 
    <span>
      7
    </span>
     are out of order
  </p>
</div>
`;

exports[`Explanation should match snapshot for default case 1`] = `
<div
  className="explanation"
>
  <p />
</div>
`;
